File: Solutions/1784/1784.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1784.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-10-3 11:13:36
***************************************************************************************************/

bool checkOnesSegment(char * s){
    bool flag = true;
    for (int i=0; i<strlen(s); i++){
        if (s[i]=='1') {
            if (!flag) return false;
        } else flag = false;
    }
    return true;
}

File: Solutions/1784/1784.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/10/3 11:17
# File: 1784.py
# Desc: 


class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        # return sum([1 if i else 0 for i in s.split('0')]) <= 1
        # return '0' not in s.strip('0')
        flag = True
        for i in s:
            if i == '1':
                if not flag: return False
            else:
                flag = False
        return True
